src/main/c/openssl/extconf.rb

```diff
@@ -30,11 +30,10 @@
 
 if ENV['OPENSSL_PREFIX']
   $CFLAGS += " -I #{ENV['OPENSSL_PREFIX']}/include"
-  $LIBS += " -l #{ENV['OPENSSL_PREFIX']}/lib/libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-else
-  $LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
 end
 
+$LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
```

Comment:
FWIW MRI uses `--with-openssl-dir` (which is the prefix). https://github.com/rbenv/ruby-build/issues/377#issuecomment-296293093